src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnectorConfig.java

```diff
@@ -0,0 +1,174 @@
+  private static final Logger LOG = LoggerFactory.getLogger(AribaConnectorConfig.class);
+  private static final String COMMON_ACTION = ResourceConstants.ERR_MISSING_PARAM_OR_MACRO_ACTION.getMsgForKey();
+
+  @Macro
+  @Description("Ariba Client ID.")
+  private final String clientId;
+
+  @Macro
+  @Description("Ariba Client Secret.")
+  private final String clientSecret;
+
+  @Macro
+  @Description("Ariba Application Key.")
+  private final String apiKey;
+
+  @Macro
+  @Description("Base Path of Ariba API.")
+  private final String baseURL;
+
+  @Macro
+  @Description("Realm name from which data is to be extracted.")
+  private final String realm;
+
+  public AribaConnectorConfig(String clientId, String clientSecret, String apiKey, String baseURL, String realm) {
+    this.clientId = clientId;
+    this.clientSecret = clientSecret;
+    this.apiKey = apiKey;
+    this.baseURL = baseURL;
+    this.realm = realm;
+  }
+
+  public String getClientId() {
+    return clientId;
+  }
+
+  public String getClientSecret() {
+    return clientSecret;
+  }
+
+  public String getApiKey() {
+    return apiKey;
+  }
+
+  public String getBaseURL() {
+    return baseURL;
+  }
+
+  public String getRealm() {
+    return realm;
+  }
+
+  /**
+   * Validates the credentials parameters.
+   *
+   * @param failureCollector {@code FailureCollector}
+   */
+  public void validateCredentials(FailureCollector failureCollector) {
+
+    if (AribaUtil.isNullOrEmpty(getClientId()) && !containsMacro(CLIENT_ID)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("Ariba Client Id");
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(CLIENT_ID);
+    }
+    if (AribaUtil.isNullOrEmpty(getClientSecret()) && !containsMacro(CLIENT_SECRET)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("Ariba Client Secret");
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(CLIENT_SECRET);
+    }
+    if (AribaUtil.isNullOrEmpty(getApiKey()) && !containsMacro(APIKEY)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("Ariba API Key");
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(APIKEY);
+    }
+    if (AribaUtil.isNullOrEmpty(getBaseURL()) && !containsMacro(BASE_URL)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey(ResourceConstants.API_END_POINT);
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(BASE_URL);
+    }
+    if (AribaUtil.isNullOrEmpty(getRealm()) && !containsMacro(REALM)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey(ResourceConstants.REALM_NAME);
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(REALM);
+    }
+  }
+
+  public void validateToken(FailureCollector collector) throws AribaException, IOException {
+    String tokenUrl = "https://api.au.cloud.ariba.com/v2/oauth/token";
```

Comment:
put it at class level and make it static final

Reply:
or we can use the method `generateTokenURL()` to get the token Url

Reply:
@vikasrathee-cs Declared it as final but not static as some of the methods used inside it are not static and also can't be declared as static.
@sgarg-CS We can't do that as there is only one way to generate it i.e by taking baseurl as input and if BaseURL itself is entered wrong then validation will not work.

Reply:
Display an error message if the BaseURL entered is wrong.